examples/integer-localizations.jl

```diff
@@ -0,0 +1,222 @@
+#################################################################################
+# A minimal implementation of the localization interface for the ring ℤ         #
+#################################################################################
+
+export FmpzComplementOfPrimeIdeal, FmpzPowersOfElement, FmpzComplementOfZeroIdeal
+export FmpzLocalizedRing, FmpzLocalizedRingElem
+export ambient_ring, localize_at, parent, numerator, denominator
+
+
+#################################################################################
+# Multiplicatively closed sets given by powers of a specific integer            #
+#################################################################################
+
+@Markdown.doc """
+FmpzPowersOfElement
+
+The multiplicative set given by the powers of some element.
+"""
+mutable struct FmpzPowersOfElement <: AbsMultSet{FlintIntegerRing, fmpz}
+  d::Vector{fmpz} # a vector of admissible denominators. 
+
+  function FmpzPowersOfElement(denominators::Vector{fmpz})
+    return new(denominators)
+  end
+end
+
+### required getter functions
+ambient_ring(S::FmpzPowersOfElement) = ZZ
+
+### required functionality
+function Base.in(a::fmpz, S::FmpzPowersOfElement) 
+  # check whether ∃ c ∈ ℤ, k ∈ ℕ₀: c ⋅ a = (∏ᵢ dᵢ)ᵏ, where dᵢ are the admissible denominators in S.
+  d = prod(denominators(S))
+  g = gcd(a, d)
+  while(!isone(g))
+    a = divexact(a, g)
+    g = gcd(a, d)
+  end
+  return isone(a)
+end
+
+### additional constructors
+FmpzPowersOfElement(d::fmpz) = FmpzPowersOfElement([d])
+FmpzPowersOfElement(d::Oscar.IntegerUnion) = FmpzPowersOfElement(ZZ(d))
```

Comment:
You already handle `fmpz` in the line before, so no need to do it again
```suggestion
FmpzPowersOfElement(d::Integer) = FmpzPowersOfElement(ZZ(d))
```

Reply:
No, not really. The first line converts to an array. The second refers to the first.